Lib/collections/__init__.py

```diff
@@ -1050,7 +1050,7 @@ def new_child(self, m=None, **kwargs):      # like Django's Context.push()
     @property
     def parents(self):                          # like Django's Context.pop()
         'New ChainMap from maps[1:].'
-        return self.__class__(*self.maps[1:])
+        return self.__class__(*_islice(self.maps, 1, None))
```

Comment:
This change doesn't do anything useful.  The time saved by not making a list is offset by the time to create the `islice` object.

On the receiving end, the `__init__()` method converts the input to a list.  This goes faster if the input is already a list than for an iterator . The former has a fast path and can take advantage of a length hint.

Reply:
Yea I think there is a better way to handle this anyway. I probably should have left it off the PR as it wasn't as heavily tested as the iter behavior.